src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,409 @@
+export const description = `
+copyImageBitmapToTexture Validation Tests in Queue.
+
+Test Coverage:
+- For source.imagBitmap:
+  - imageBitmap generated from ImageData:
+    - Check that an error is generated when imageBitmap is closed. Otherwise, no error should be
+      generated.
+
+- For destination.texture:
+  - For 2d destination textures:
+    - Check that an error is generated when texture is in destroyed state. Otherwise, no error should
+      be generated.
+    - Check that an error is generated when texture is created without usage COPY_DST. Otherwise,
+      no error should be generated.
+    - Check that an error is generated when sample count is not 1. Otherwise, no error should be
+      generated.
+    - Check that an error is generated when mipLevel is too large. Otherwise, no error should be
+      generated.
+    - Check that an error is generated when texture format is not valid. Otherwise, no error should
+      be generated.
+
+- For copySize:
+  - Check that an error is generated when copySize has 0 in width, height or depth. Otherwise,
+    no error should be generated.
+  - Check that an error is generated when destination.texture.origin + copySize is too large.
+    Otherwise, no error should be generated.
+
+TODO: more test to cover source.imageBitmap generated from different source.
+TODO: 1d and 3d texture
+`;
+
+import { poptions, params } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { unreachable } from '../../../../common/framework/util/util.js';
+import { kAllTextureFormats, kTextureUsages } from '../../../capability_info.js';
+import { ValidationTest } from '../validation_test.js';
+
+const DEFAULT_BYTES_PER_PIXEL = 4;
+const DEFAULT_WIDTH = 4;
+const DEFAULT_HEIGHT = 4;
+const DEFAULT_DEPTH = 4;
+const DEFAULT_MIPLEVEL_COUNT = 1;
+
+const kValidTextureFormatsForCopyIB2T = [
+  'rgba8unorm',
+  'rgba8unorm-srgb',
+  'bgra8unorm',
+  'bgra8unorm-srgb',
+  'rgb10a2unorm',
+  'rgba16float',
+  'rgba32float',
+  'rg8unorm',
+  'rg16float',
+];
+
+interface WithDimension {
+  dimension: GPUTextureDimension;
+}
+
+// Helper function to generate correct texture size.
+function copySizeTestForDstTextureSize({ dimension }: WithDimension) {
```

Comment:
Why not just GPUTextureDimension?

Reply:
(Similar interface used in new commit)
The function is used in .expand template, which need this interface.